chromium_src/components/signin/public/identity_manager/identity_manager_builder.cc

```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "../../../../../../components/signin/public/identity_manager/identity_manager_builder.cc"
+
+#include "brave/components/signin/internal/identity_manager/brave_primary_account_mutator_impl.h"
+#include "brave/components/signin/public/identity_manager/brave_identity_manager.h"
+
+namespace signin {
+
+IdentityManager::InitParameters BuildBraveIdentityManagerInitParameters(
```

Comment:
Is this function different from Chromium's `BuildIdentityManagerInitParameters`?

Reply:
yes, the only difference is at 
```
  init_params.primary_account_mutator =
      std::make_unique<BravePrimaryAccountMutatorImpl>(
          account_tracker_service.get(), primary_account_manager.get(),
          params->pref_service);
```

, we are constructing `BravePrimaryAccountMutatorImpl` instead of `PrimaryAccountMutatorImpl`

Reply:
Can we just `#define PrimaryAccountMutatorImpl BravePrimaryAccountMutatorImpl` and use their function?

Reply:
^that is a good idea to get rid of the other duplicate parts

Reply:
We must to keep the ability to create pure Chromium IdentityManager, it is required for some extensions, and use Brave-stubbed only for `ProfileSyncManager`. And `#define PrimaryAccountMutatorImpl BravePrimaryAccountMutatorImpl` will change that for global scope.

I had tried to put use double-include  with define, 

```
#include "../../../../../../components/signin/public/identity_manager/identity_manager_builder.cc"

#define PrimaryAccountMutatorImpl BravePrimaryAccountMutatorImpl

#include "../../../../../../components/signin/public/identity_manager/identity_manager_builder.cc"

#undef PrimaryAccountMutatorImpl
```

But that had made a mess with lots of compile errors.

Reply:
what about we call `BuildIdentityManagerInitParameters` and change `primary_account_mutator` from  return value of it in `BuildBraveIdentityManagerInitParameters`?

Reply:
Thanks, this works.

Some code required for `primary_account_mutator` is still duplicated, but significantly less amount.

Fixed at https://github.com/brave/brave-core/pull/7346/commits/5b4658bbf4189d40a5b1a84365e74c5eef5ce70e

Reply:
Do we need to replicate those? Both original `account_tracker_service` and `primary_account_manager` are in the returned `init_params`.

Reply:
indeed, we don't need to replicate all those, fixed in https://github.com/brave/brave-core/pull/7346/commits/dc2d44b5911633d1d30ab050c5b0b70212e80b72

Reply:
Awesome! Just as a nit: I wouldn't even declare this function in `.h` and put it into an anonymous namespace here, since it's only called from this `.cc`

Reply:
thanks @mkarolin @darkdh